python/src/spark_rapids_ml/core.py

```diff
@@ -1057,9 +1057,17 @@ def _fit_internal(
 
         return models
 
-    def _fit(self, dataset: DataFrame) -> "_CumlModel":
+    def _fit(self, dataset: DataFrame) -> "Model":
         """fit only 1 model"""
-        return self._fit_internal(dataset, None)[0]
+        if self._pyspark_class() and self._fallback_enabled and self._cpu_fallback():
+            logger = get_logger(self.__class__)
+            logger.warning("Invoking cpu estimator fit().")
+            cpu_class = self._pyspark_class()
+            cpu_est = cpu_class(**{p.name: v for p, v in self.extractParamMap().items() if self.isSet(p) and hasattr(cpu_class, p.name)})  # type: ignore
+            model = cpu_est.fit(dataset)
```

Comment:
Nit: do we want to print out a logging message when falling back?

Reply:
Done a few lines previous.